core/sail/lucene/src/main/java/org/eclipse/rdf4j/sail/lucene/LuceneDocument.java

```diff
@@ -140,13 +154,33 @@ public boolean hasProperty(String fieldName, String value) {
 	public void addGeoProperty(String field, String value) {
 		LuceneIndex.addStoredOnlyPredicateField(field, value, doc);
 		try {
-			SpatialStrategy geoStrategy = geoStrategyMapper.apply(field);
-			Shape shape = geoStrategy.getSpatialContext().readShapeFromWkt(value);
-			for (IndexableField f : geoStrategy.createIndexableFields(shape)) {
-				doc.add(f);
+			String wkt = value;
+			// wkt = wkt.replace("\"", "").replace("^^<http://www.opengis.net/ont/geosparql#wktLiteral>", "");
+			// System.out.println(wkt);
+			Object shape = SimpleWKTShapeParser.parse(wkt);
+			if (shape instanceof Polygon[]) {
+				for (Polygon p : (Polygon[]) shape) {
+					for (Field f : LatLonShape.createIndexableFields(field, p)) {
+						doc.add(f);
+					}
+				}
+			} else if (shape instanceof Polygon) {
+				for (Field f : LatLonShape.createIndexableFields(field, (Polygon) shape)) {
+					doc.add(f);
+				}
+			} else if (shape instanceof Line) {
+				for (Field f : LatLonShape.createIndexableFields(field, (Line) shape)) {
+					doc.add(f);
+				}
+			} else if (shape instanceof double[]) {
+				double point[] = (double[]) shape;
+				// System.out.println(point[0]+" "+point[1]);
+				doc.add(new LatLonPoint(field, point[1], point[0]));
+			} else {
+				throw new IllegalArgumentException("Geometry is not supported");
 			}
-		} catch (ParseException e) {
-			// ignore
+		} catch (Exception e) {
```

Comment:
Let's be specific about the exceptions we're catching here:

     catch (ParseException | IllegalArgumentException e) {